DESCRIPTION

```diff
@@ -81,6 +81,7 @@ Collate:
     'condition_message_linter.R'
     'conjunct_test_linter.R'
     'consecutive_assertion_linter.R'
+    'consecutive_mutate_linter.R'
```

Comment:
Should this linter be named (e.g. `unnecessary_consecutive_calls_linter()`) in a way to anticipate further cases like this in the _tidyverse_ workflow?

E.g. I could imagine a similar linter for `dplyr::filter()`

``` r
library(dplyr, warn.conflicts = FALSE)

df <- tibble(a = 1:4, b = 4:1)

identical(
  df %>%
    filter(a > 2) %>%
    filter(b < 3),
  df %>%
    filter(a > 2 & b < 3)
)
#> [1] TRUE
```

<sup>Created on 2023-11-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Note that this is different from `conjunct_test_linter()`, which is not about successive function calls.

Reply:
Careful:

`df %>% group_by(x) %>% filter(y == 42L) %>% filter(n() > 1L)` cannot be collapsed into a single filter call, because the first filter changes group sizes.

Reply:
Yes, exactly why this case was not considered. It's impossible to tell statically whether this is happening, too.

Reply:
(historical footnote: we originally had a `ConsecutiveCallLinter()` that covered `suppressMessages()`, `mutate()`, and `stopifnot()`, but each case drifted far enough apart that we gave them their own linter)

Reply:
Hadn't thought about this. Thanks for this insight!

``` r
library(dplyr)
library(palmerpenguins)

setdiff(
  penguins %>%
    group_by(sex) %>%
    filter(island == "Dream", n() > 1L),
  penguins %>%
    group_by(sex) %>%
    filter(island == "Dream") %>%
    filter(n() > 1L)
)
#> # A tibble: 1 × 8
#> # Groups:   sex [1]
#>   species island bill_length_mm bill_depth_mm flipper_length_mm body_mass_g
#>   <fct>   <fct>           <dbl>         <dbl>             <int>       <int>
#> 1 Adelie  Dream            37.5          18.9               179        2975
#> # ℹ 2 more variables: sex <fct>, year <int>
```

<sup>Created on 2023-11-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Basically, we either need to have `filter(x, A, B, C)` equivalent to `filter(x, A && B && C)` or `x |> filter(A) |> filter(B) |> filter(C)`; {dplyr} chooses the former.